### PR TITLE
Feat/test for peer_shard_info_v2 message

### PR DESCRIPTION
**Pull request by lwojtow**
1. Add TmPeerShardInfoV2 payload
2. Add optional history sharding
3. Add TmGetPeerShardInfoV2 message test

### CODE REVIEW

File: src/setup/config.rs [hunk 1]
```diff
@@ -129,6 +129,20 @@ impl RippledConfigFile {
...
+            writeln!(&mut config_str, "[shard_db]")?;
+            writeln!(
+                &mut config_str,
+                "path={}",
+                path.join(RIPPLED_DIR)
+                    .join("db/shard/nudb")
+                    .to_str()
+                    .unwrap()
+            )?;
+            writeln!(&mut config_str, "max_historical_shards=12")?;
```
Comment (ljedrz) [inline src/setup/config.rs:142]: we might want to comment on / link to the reasoning behind this value

File: src/tests/conformance/query/peer_shard_info.rs [hunk 2]
```diff
@@ -97,3 +97,47 @@ async fn check_relay_for_key_type(key_type: u8, relays: u32) {
...
+    synth_node
+        .connect(node.addr())
+        .await
+        .expect("unable to connect");
+
+    // Create a payload with a valid key.
+    let mut key = vec![PUBLIC_KEY_TYPES[0]]; // Place the key type as the first byte.
+    key.resize(PUBLIC_KEY_LENGTH, 0x1); // Append 32 bytes serving as a dummy public key.
+    let payload = Payload::TmGetPeerShardInfoV2(TmGetPeerShardInfoV2 {
+        peer_chain: vec![TmPublicKey { public_key: key }],
```
Comment (ljedrz) [inline src/tests/conformance/query/peer_shard_info.rs:126]:
```suggestion
    let mut public_key = vec![PUBLIC_KEY_TYPES[0]]; // Place the key type as the first byte.
    public_key.resize(PUBLIC_KEY_LENGTH, 0x1); // Append 32 bytes serving as a dummy public key.
    let payload = Payload::TmGetPeerShardInfoV2(TmGetPeerShardInfoV2 {
        peer_chain: vec![TmPublicKey { public_key }],
```
nit

**Review by ljedrz — APPROVED** [review]
LGTM with some minor comments

**Review by Rqnsom — APPROVED** [review]
LGTM 👍

**Pull request merged**